db/seed.go

```diff
@@ -0,0 +1,217 @@
+package db
+
+import (
+	"fmt"
+	"golang.org/x/crypto/bcrypt"
+	"os"
+	"time"
+)
+
+func (s *PostgresStore) Seed() error {
+	tables := []string{"entry", "account_membership", "membership", "account", "event"}
+	var err error
+	if err = s.cleanup(tables); err != nil {
+		return err
+	}
+
+	if err = s.createEventType(); err != nil {
+		return err
+	}
+	if err = s.createRole(); err != nil {
+		return err
+	}
+	if err = s.createAccountTable(); err != nil {
+		return err
+	}
+	if err = s.createEventTable(); err != nil {
+		return err
+	}
+	if err = s.createMembershipTable(); err != nil {
+		return err
+	}
+	if err = s.createAccountMembershipTable(); err != nil {
+		return err
+	}
+	if err = s.createEntryTable(); err != nil {
+		return err
+	}
+
+	now := time.Now()
+	startOfWeek := now.AddDate(0, 0, 1-int(now.Weekday()))
+	var events []Event
+	for i := -3; i < 16; i++ {
+		dayStart := startOfWeek.AddDate(0, 0, i)
+		startTime := time.Date(dayStart.Year(), dayStart.Month(), dayStart.Day(), 16, 0, 0, 0, time.Local)
+		endTime := startTime.Add(time.Hour)
+		event := Event{
+			Type:     OPEN_GYM,
+			Title:    fmt.Sprintf("Event at %s", startTime),
+			Start:    startTime,
+			End:      endTime,
+			Capacity: 10,
+			Price:    100,
+		}
+		events = append(events, event)
+	}
+	for _, event := range events {
+		if _, err = s.CreateEvent(&event); err != nil {
+			return err
+		}
+	}
+
+	memberships := []Membership{
+		{1, OPEN_GYM, 30, 30, 1000, time.Now(), nil},
+		{2, LECTURE, 30, 30, 1500, time.Now(), nil},
+		{3, ALL, 30, 30, 2000, time.Now(), nil},
+	}
+	for _, membership := range memberships {
+		if _, err = s.CreateMembership(&membership); err != nil {
+			return err
+		}
+	}
+	err = s.createAdmin()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *PostgresStore) cleanup(tables []string) error {
+	for i := 0; i < len(tables); i++ {
+		query := fmt.Sprintf(`drop table if exists %s`, tables[i])
+		if _, err := s.Db.Exec(query); err != nil {
+			return err
+		}
+	}
+	_, err := s.Db.Exec(`drop type if exists event_type`)
+	if err != nil {
+		return err
+	}
+	_, err = s.Db.Exec(`drop type if exists role`)
+	return err
+}
+
+func (s *PostgresStore) createAccountTable() error {
+	query := `create table if not exists account (
+		id serial primary key,
+		first_name varchar(255),
+		last_name varchar(255),
+		encrypted_password varchar(255),
+		email varchar(255) unique,
+		credit int,
+		role role default 'user',
+		created_at timestamp default current_timestamp,
+		deleted_at timestamp default null
+	)`
+	_, err := s.Db.Exec(query)
+	return err
+}
```

Comment:
Personally, I would much rather have this initial schema files in separate directory in SQL files and I would use some migrator tool for initializing the database on app startup. For example, the [Goose](https://github.com/pressly/goose) exists.